File: Difficulty: Basic/Divisor/divisor.py
#{ 
 # Driver Code Starts
#Initial Template for Python 3

# } Driver Code Ends
#User function Template for python3
n = int(input())

# List to store divisors
divisors = []

# Iterate from 1 to sqrt(n)
for i in range(1, int(n**0.5) + 1):
    if n % i == 0:  # Check if i is a divisor
        divisors.append(i)  # Add i to divisors list
        if i != n // i:  # To avoid adding the square root twice when n is a perfect square
            divisors.append(n // i)

# Sort the divisors and print them
divisors.sort()
print(" ".join(map(str, divisors)))


#{ 
 # Driver Code Starts.

# } Driver Code Ends

File: Difficulty: Basic/Divisor/divisor.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.Scanner;

class Main {

    
// } Driver Code Ends

// User function Template for Java

public static void divisor(int number) {
    
    // Loop through all numbers from 1 to 'number' to check for divisibility
    for (int potentialDivisor = 1; potentialDivisor <= number; potentialDivisor++) {
        
        // If 'number' is divisible by 'potentialDivisor', print it
        if (number % potentialDivisor == 0) {
            System.out.print(potentialDivisor + " ");
        }
    }
}


//{ Driver Code Starts.

    public static void main(String[] args)
    {
        Scanner scn = new Scanner(System.in);
        int t = scn.nextInt();
        while(t-- > 0) {
            int n = scn.nextInt();
            divisor(n);
            System.out.println();
        
System.out.println("~");
}
        scn.close();
    }
}
// } Driver Code Ends